### PR TITLE
Add repositories dropdown list [SCI-4506]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-4506](https://biosistemika.atlassian.net/browse/SCI-4506)

### What was done
Add repositories dropdown list

### CODE REVIEW

File: app/views/my_modules/protocols.html.erb [hunk 1]
```diff
@@ -41,7 +41,20 @@
...
+      <div class="actions-block">
+        <div class="dropdown repositories-assign-container"
+             data-repositories-url="<%= dropdown_list_my_module_my_module_repositories_path(@my_module) %>"
+        >
+          <a href="#" id="repositories-assign-button" class="btn btn-secondary btn-block" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <span class="fas fa-file-signature"></span>
+            <span><%= t('my_modules.assigned_items.assign_from') %></span>
+            <span class="caret pull-right"></span>
+          </a>
+          <div class="dropdown-menu repositories-dropdown-menu" aria-labelledby="repositories-assign-button">
```
Comment (mlorb) [inline app/views/my_modules/protocols.html.erb:54]: Why don't you use `ul` and `li` elements? (Bootstrap standard)
The hover style is missing.

File: app/controllers/my_module_repositories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class MyModuleRepositoriesController < ApplicationController
+  before_action :load_my_module, only: :dropdown_list
+  before_action :check_my_module_view_permissions, only: :dropdown_list
+
+  def dropdown_list
+    @repositories = Repository.accessible_by_teams(current_team)
```
Comment (mlorb) [inline app/controllers/my_module_repositories_controller.rb:8]: You forgot on `Repository.used_on_task_but_unshared`. You can check current implementation of inventories dropdown in `navbar-secondary`

File: app/assets/stylesheets/my_modules/protocols/index.scss [hunk 3]
```diff
@@ -37,6 +37,57 @@
       margin-top: 11px;
     }
   }
+
+  .task-section-header {
+    display: flex;
```
[the remark this answers is not in the document]
Comment (aignatov-bio) [reply app/assets/stylesheets/my_modules/protocols/index.scss:42]: Fixed

File: app/views/my_modules/protocols.html.erb (hunk 1 above)
Comment (aignatov-bio) [reply app/views/my_modules/protocols.html.erb]: Fixed

File: app/controllers/my_module_repositories_controller.rb (hunk 2 above)
Comment (aignatov-bio) [reply app/controllers/my_module_repositories_controller.rb:38]: I will be in separate task

File: app/controllers/my_module_repositories_controller.rb [hunk 4]
```diff
@@ -30,17 +32,29 @@ def full_view_table
...
 
-  def load_vars
+  def load_my_module
     @my_module = MyModule.find(params[:my_module_id])
+    render_404 unless @my_module
+  end
+
+  def load_repository
     @repository = Repository.find(params[:id])
+    ender_404 unless @repository
```
Comment (okriuchykhin) [inline app/controllers/my_module_repositories_controller.rb:50]: find throws an exception if object is not found

Comment (okriuchykhin) [reply app/controllers/my_module_repositories_controller.rb:50]: and there is typo

**Pull request merged**